addons/hr_expense/wizard/account_register_payments.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, _
+from werkzeug import url_encode
+
+class AccountRegisterPayments(models.TransientModel):
+    _inherit = "account.register.payments"
+
+    @api.model
+    def default_get(self, fields):
+        res = super(AccountRegisterPayments, self).default_get(fields)
+        context = dict(self._context or {})
+        active_model = context.get('active_model')
+        if active_model != 'hr.expense.sheet':
+            return res
+        active_ids = context.get('active_ids', [])
+        expense_sheet = self.env[active_model].browse(active_ids)
+        partner_id = expense_sheet.address_id.id or expense_sheet.employee_id.id and expense_sheet.employee_id.address_home_id.id
+        res.update(
+            partner_type='supplier',
+            payment_type='outbound',
+            amount=abs(expense_sheet.total_amount),
+            currency_id=expense_sheet.currency_id.id,
+            partner_id=partner_id
+        )
+        return res
+
+    def get_payment_vals(self):
```

Comment:
@qdp-odoo ```account.register.payments``` model is been refactored a bit by this commit https://github.com/odoo/odoo/commit/4ffb61db1d3228c9f21f2acc2d71493b1b07983b and we have to adapt our code accordingly, could you please have a look at new changes and share your opinion ?
Note: we have to split ```create_payment()``` of ```account.register.payments``` as we needed to perform reconciliation of expense payments and also need to log the payment reference in expenses(please see https://github.com/odoo/odoo/pull/17267/files#diff-0fe44177cffd5b8c1f0d890f55b505c2R43).